packages/ui/src/composites/wui-otp/index.ts

```diff
@@ -0,0 +1,144 @@
+import { html, LitElement } from 'lit'
+import { customElement } from 'lit/decorators.js'
+import { resetStyles } from '../../utils/ThemeUtil'
+import styles from './styles'
+import '../../layout/wui-flex'
+import '../wui-input-numeric'
+import { ref } from 'lit/directives/ref.js'
+
+const regex = {
+  number: /^[0-9]+$/u
+}
+
+@customElement('wui-otp')
+export class WuiOtp extends LitElement {
+  public static styles = [resetStyles, styles]
+
+  // -- Render -------------------------------------------- //
+  public render() {
+    return html`
+      <wui-flex gap="s">
+        <wui-flex gap="xxs">
+          ${[0, 1, 2].map(
+            index => html`
+              <wui-input-numeric
+                @input=${(e: InputEvent) => this.handleInput(e, index)}
+                @keydown=${(e: KeyboardEvent) => this.handleKeyDown(e, index)}
+                ${ref(el => this.registerInputRef(el as HTMLElement, index))}
+              ></wui-input-numeric>
+            `
+          )}
+        </wui-flex>
+        <wui-flex gap="xxs">
+          ${[3, 4, 5].map(
+            index => html`
+              <wui-input-numeric
+                @input=${(e: InputEvent) => this.handleInput(e, index)}
+                @keydown=${(e: KeyboardEvent) => this.handleKeyDown(e, index)}
+                ${ref(el => this.registerInputRef(el as HTMLElement, index))}
+              ></wui-input-numeric>
+            `
+          )}
+        </wui-flex>
+      </wui-flex>
+    `
+  }
+
+  // -- Private ------------------------------------------- //
+  private inputs: HTMLInputElement[] = []
+
+  private validate = (character: string) => regex.number.test(character)
```

Comment:
`regex.number.test(character)` bit should probably be a core utility i.e `isNumber` or even `isPositiveNumber`